web/yo/app/scripts/app.js

```diff
@@ -183,20 +183,19 @@ angular.module('oncokbApp').run(
                 var hugoSymbol = '';
                 if (fromIndex !== -1) {
                     //When the curator left the gene page
-                    hugoSymbol = window.location.href.substring(fromIndex+6);
+                    hugoSymbol = window.location.href.match(/\/([^\/]+)\/?$/)[1];
                     window.localStorage.geneName = hugoSymbol;
                 }
                 var toIndex = $location.path().indexOf('/gene/');
                 if (toIndex !== -1) {
                     //When the curator enter the gene page
-                    hugoSymbol = $location.path().substring(toIndex+6);
+                    hugoSymbol = $location.path().match(/\/([^\/]+)\/?$/)[1];
```

Comment:
Let's make this regular expression a constant, it can be used in these three places. Or even add a util to parse the URL.